### ISSUE TITLE
Sound server keeps crashing/restarting....causing lag spikes.

### ISSUE THREAD

**Issue by xAlicatt** [OreCruncher/DynamicSurroundings#107]
*Mod Version:* 3.4.5.3

*Forge Version:* 2281 

*Link to crash log (if applicable):* not applicable...no crash log to report, though I have had the client just outright crash after one of these lag spikes, but there is no crash report in the folder for it

*Description:* Since the update to the new build of DS, we have been having our sound servers crash a lot. By default it's set to restart it, which causes a lag spike. One member has reported being spammed with the sound server crashing/restarting error message in game, that is only remedied for a short time by restarting the client.

I have disabled auto restart of the sound server in settings, but I'm still getting the lag spikes that seem to be caused by sound issues. Still seems like the sound server is auto restarting itself, even though I've asked it not to.

Is there some specific settings you can recommend disabling/turning down/etc to eliminate these sound crashes?

**Comment by OreCruncher** [comment]
If you turn off the auto restart you will get text in the client window about every 30 seconds indicating the that client should be restarted.  When the auto-restart option is turned off, what messages show up in chat?  It will tell me what logic path is being followed.

Also, what other sound based mods do you have installed?

**Comment by OreCruncher** [comment]
@xAlicatt Following up - do you have the text that is being sent to chat?  I made a tweak in v3.4.5.4 that will only send the restart notice ONCE if the auto-restart function is disabled.  Also, what version of DS were you using prior to the update?

**Comment by xAlicatt** [comment]
We ended up rolling back to the previous release we have been using up to this point. I actually prefer the older release anyway as far as sound goes. I like to live in sacred  springs, and the constant crickets and swamp sounds started to wear on me after a while, even with the volume turned down to like 3.

In any case, I do miss the updated config options, but I can deal. 

My pack is on Twitch (curse} jakspack-1-10-2  

When I get home, I will fork the latest version and update ds to see if the problems still exist. I think Avaritia and More Player Models may have been the cause of much of our problems, as we have had little issues since removing them... Both client and server side. 

I'll report back after I do this.

**Comment by OreCruncher** [comment]
I have your pack installed and have yet to trigger the problem.  Have you noticed any set of circumstances that are common between these events?

**Comment by xAlicatt** [comment]
The most recent version of my pack is running the old release of DS, so you won't run into the problems.

Yes, I would rather not have the Jungle sounds in Sacred Springs...I don't really feel like it fits at all. Forest sounds during the day and crickets at night like they were in the older release are more appropriate imo. 

I will look into removing Fairy Lights...i love the idea of the mod, but if it will make that much of a difference, i will definitely drop it.

**Comment by OreCruncher** [comment]
@xAlicatt In my testing I am using the most current build of DS, not the one in the pack.

**Comment by xAlicatt** [comment]
Good to know...I'll give it a try myself. Can I turn off the jungle sounds in SS biomes myself? I've tried to figure out how to do it, but gave up :P

**Comment by OreCruncher** [comment]
@xAlicatt External config files can override what DS does with the biome.  What I am going to do is add additional information to my internal config for BoP to override the generic settings for Sacred Spring and set the ones for Forest.  Need to do a little research for it.  Should be in the next release (v3.4.5.5).

**Comment by xAlicatt** [comment]
@OreCruncher Sounds good. :) Thank you for the help!

**Comment by OreCruncher** [comment]
@xAlicatt BTW, I am curious.  You turned particle effects and range way down.  Did you do this because of lag?  Also, when did you make these changes?  The reason I ask is that I did a lot of work to get particle performance up and if you haven't checked it out in a while you may want to.

**Comment by xAlicatt** [comment]
@orecruncher yes I play on a 3 year old entertainment laptop...so particles tend to cause issues. Rain, particularly, causes fps issues, especially in a mega - modded environment like I tend to prefer playing in. 

I try to get the best graphical experience I can without sacrificing too much fps...since i know as soon as i start adding machines and pipes and decorative models to the game environment, I will definitely be taking an fps hit. For instance, I like the look of the fancy leaves, even the very fancy leaves of the foliage mod.  But,  since I tend to prefer building in leaf-heavy biomes like BOPs Sacred Springs, Rainforest, Tropical Rainforest, Bamboo Forest, Overgrown Peaks, vanilla Roofed Forest, etc... I don't run the foliage mod, and I turn off fancy graphics.

**Comment by xAlicatt** [comment]
@orecruncher I think I turned down all of that when I was trying to pinpoint the lag spike problems we were having, but I think that a couple of the mods I mentioned above are the main cause of the issue we were having. I was able to update to the last version of DS in my pack last night, and noticed no significant problems in doing so. I also think that the removal of Fairy Lights (than you for that tip btw) has raised my fps by 10-20.

**Comment by OreCruncher** [comment]
Glad I could help with the Fairy Lights.  Like you I think the mod would be cool in a pack, but the hit on performance is pretty large when dealing with a sink pack.

Another point of lag is sound.  When I reworked the rain particle handling I reduced the amount of sound events generated by a large degree.  Also, I put in sound culling to deal with frequent sounds - like animal noises.  What was unexepcted is that flowing water and lava is really noisy, especially around lava lakes.  I cull liquid sounds and it reduces the impact on the client.

Oh, one more thing.  There is an option in the latest version to kill off the suspend particle effect.  Suspend particles are those little dark motes that spawn in water blocks.  If you or your players live around large bodies of water and have particle lag killing those particles could help.
